vowpalwabbit/parse_regressor.cc

```diff
@@ -316,6 +316,12 @@ void save_load_header(
         }
       }
 
+      if (!read || all.model_file_ver >= VERSION_FILE_WITH_RANDOM_SEED)
```

Comment:
I like that this is going into the general save_load header but I am concerned that this might break forward compatibility on models and so might be safer to put this into the cats reduction where the seed is being used

Reply:
I had this fix in a lower reduction initially to solve the bug, however the random seed is used in other places like ccb, so the thinking here was that including it in the VW model would avoid other current / future bugs